jvm/src/Language/Java.hs

```diff
@@ -416,6 +427,9 @@ class (Interp (Uncurry a) ~ ty, SingI ty, IsReferenceType ty)
       => Reflect a ty where
   reflect :: a -> IO (J ty)
 
+  default reflect :: Coercible a ty => a -> IO (J ty)
+  reflect x = newLocalRef (jobject x)
+
```

Comment:
There's duplication in the Reify and Reflect instances for  `J ty` which should probably use the default now.

That said, I'm not convinced the user should be given instances of `Reify (J ty) ty` and `Reflect (J ty) ty` unless he asks for them.